src/node_modules/_file-explorer.js

```diff
@@ -0,0 +1,292 @@
+    'src/leasing.sol': '',
+    'src/gmbh/contract.sol': false,
+    'src/gmbh/test.sol': false,
+    'src/gmbh/company.sol': false,
+    'src/gmbh/node_modules/ballot.sol': false,
+    'src/ug/finance.sol': false,
+    'app/solidity/mode.sol': true,
+    'app/ethereum/constitution.sol': true
+  }
+  Object.keys(sources).forEach(function (key) { files.set(key, sources[key]) })
+
+  var deleteButton = yo`<span class=${css.remove} onclick=${deletePath}><i class="fa fa-trash" aria-hidden="true"></i></span>`
+
+  // files.event.register('fileChanged', function (filepath) { })
+  files.event.register('fileAdded', function (filepath) {
+    console.log('fileAdded', filepath)
+  })
+  files.event.register('fileFocus', function (path) {
+    if (filepath == path) return
+    filepath = path
+    var el = getElement(filepath)
+    var tmp = el
+    while ((tmp = tmp.parentElement.parentElement) && tmp.tagName === 'LI') {
+      var caret = tmp.firstChild.firstChild
+      if (caret.classList.contains('fa-caret-right')) caret.click() // expand
+    }
+    el.click() // focus element
+  })
+  files.event.register('fileRemoved', function (filepath) {
+    console.log('fileRemoved', filepath)
+  })
+  files.event.register('fileRenamed', function (oldName, newName) {
+    var el = getElement(oldName)
+    if (el) {
+      oldName = oldName.split('/')
+      newName = newName.split('/')
+      var index = oldName.reduce(function (idx, key, i) {
+        return oldName[i] !== newName[i] ? i : idx
+      }, undefined)
+      var newKey  = newName[index]
+      var oldPath = oldName.slice(0,index+1).join('/')
+      el = getElement(oldPath)
+      el.firstChild.lastChild.firstChild.innerText = newKey
+    }
+  })
+
+  var filenames = files.list()
+  var data = { }
+  Object.keys(filenames).forEach(file => setval(data, file, true))
+  var filepath = null
+  var focusElement = null
+  var textUnderEdit = null
+
+  var tv = new Treeview({
+    extractData: function (subtree, tree, key) {
+      return {
+        children: subtree instanceof Array ? subtree.map((item, index) => ({
+          key: index, value: item
+        })) : subtree instanceof Object ? Object.keys(subtree).map(key => ({
+          key: key, value: subtree[key]
+        })) : undefined
+      }
+    },
+    formatSelf: function (key, data) {
+      return yo`<label class=${data.children ? css.folder : css.file}
+        onload=${function (el) { adaptEnvironment(el, focus, hover) }}
+        onunload=${function (el) { unadaptEnvironment(el, focus, hover) }}
+        onclick=${editModeOn}
+        onkeydown=${editModeOff}
+        onblur=${editModeOff}
+      >${key}</label>`
+    }
+  })
+
+  container.appendChild(tv.render(data))
+
+  setTimeout(function () { // infere first active element from tab bar
+    var activePath = document.querySelector('.file.active').firstChild.innerText
+    var el = getElement(activePath)
+    focus.call(el,{})
+  },0)
+
+  var api = { }
+
+  return api
```

Comment:
Would it be possible reorganising the file so that the helpers aren't after the return statement? I know it is valid, but doesn't feel intuitive.